test/perceptual/consistent-evaluation-lcb.visual-diff.js

```
@@ -29,6 +28,31 @@ describe('d2l-consistent-evaluation', () => {
 		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
 	});
 
+	it('renders learner profile card', async function() {
+		await page.$eval('#open-card', async(elem) => {
+			const lcb = elem.querySelector('d2l-consistent-evaluation-learner-context-bar');
+			const userContext = lcb.shadowRoot.querySelector('d2l-consistent-evaluation-lcb-user-context');
+			const card = userContext.shadowRoot.querySelector('d2l-labs-user-profile-card');
+			card._userProfileCardSettings = {
```

Scope of the visual-diff test looks good!  Isn't just testing the card (which should in theory be covered by the visual-diff tests in the profile card repo) but you're testing a smaller piece than "the whole page" which is going to update a lot.

Setting the card settings like this is definitely fragile - ideally you'd configure this through the HM data you're passing into the test, so that changes to the card's underlying data structure and code wouldn't break this test.  But I don't know how easy/tricky that is

---

The component magically requests the instance-based settings internally. We never actually pass that information in to it, it just knows based on where it lives, where it needs to go looking. So running it in a non-lms environment, it will always fail-closed and assume we want to show the absolute minimum (basically just the username and contact links). It's possible that testing for that minimum is fine, since any issues relating to them would get snapped up in the profile card tests? 🤔

---

Hmm, can we mock out the secret automagic route it's calling at least?  You still know too much about the component's internals, but at least you'd only need to update things if the data/api side changed, not the component code?
Mostly just calling this out as a use-at-your-own-risk strategy, the ROI might not be worth implementing my suggestion

---

After some poking it's simpler to test it without passing anything. The fallback hides a bunch of information but does explode, and the effects of that only matter on the actual profile card - so we don't need to bother with it here. :) Simplified things.